### ISSUE TITLE
Provide an option to disable CUDA_GCC_VERSIONS

### ISSUE THREAD

**Issue by hmaarrfk** [pytorch/pytorch#81680]
### 🚀 The feature, motivation and pitch

I really understand the motivations of recent additions to CUDA_GCC_VERSIONS that were done in
* https://github.com/pytorch/pytorch/commit/86deecd7be248016d413e9ad6f5527a70c88b454
* https://github.com/pytorch/pytorch/pull/63230

I do think they are a step in the right direction toward improving the situation (xref https://github.com/pytorch/pytorch/issues/55267)

I'm wondering if you could add an option, as an environment variable, for us to ignore it. The motivation is that I believe that it is  possible to patch up the GCC 10 compiler with a small patch to help nvidia compatibility.

https://github.com/conda-forge/ctng-compilers-feedstock/blob/main/recipe/meta.yaml#L22=

### Alternatives

I can patch things out.... However, I would rather have the override done by other packagers that assert they know what they are doing.

You can also patch things out with a sed command
```
sed -i '/ (MINIMUM_GCC_VERSION,/d' ${SP_DIR}/torch/utils/cpp_extension.py
```

### Additional context

The other motivation is that soon enough, the GCC 11 compilers should be compatible with nvidia.

I also believe that 10.4 is compatible with the nvcc

**Comment by hmaarrfk** [comment]
A more permanent link to the relevant patch https://github.com/conda-forge/ctng-compilers-feedstock/pull/88

**Comment by rgommers** [comment]
I think it's reasonable to support this via an env var. Although the alternative of letting conda-forge carry the patch is also not much work.

**Comment by malfet** [comment]
Hmm, I believe there is already an environment variable to bypass that: `TORCH_DONT_CHECK_COMPILER_ABI`:
https://github.com/pytorch/pytorch/blob/7520f8758e4826ca66aa8dcf58f25c6730568c1f/torch/utils/cpp_extension.py#L379-L382

**Comment by rgommers** [comment]
Yes indeed, that should do the job. Not sure how I overlooked that, thanks @malfet. It has been there for a long time already.

@hmaarrfk does that work for you?

**Comment by hmaarrfk** [comment]
Very likely. I can test next week. Feel free to close if you are confident.
